airflow/models.py

```diff
@@ -201,10 +202,11 @@ def get_dag(self, dag_id):
                     dag.last_loaded < orm_dag.last_expired
                 )
         ):
-            # Reprocessing source file
+            # Then reprocess source file
             found_dags = self.process_file(
                 filepath=orm_dag.fileloc, only_if_updated=False)
 
+            # If the source file now does not export `dag_id`, delete it from self.dags
```

Comment:
I included "now" because the source file must have previously exported `dag_id` for the `del` to be run (I think)